packages/@react-spectrum/utils/src/BreakpointProvider.tsx

```diff
@@ -62,7 +62,16 @@ export function useMatchedBreakpoints(breakpoints: Breakpoints): string[] {
     }
 
     let onResize = () => {
-      setBreakpoint(getBreakpointHandler());
+      const breakpointHandler = getBreakpointHandler();
+
+      setBreakpoint(previousBreakpointHandler => {
+        if (previousBreakpointHandler.length !== breakpointHandler.length ||
+          previousBreakpointHandler.some((breakpoint, idx) => breakpoint !== breakpointHandler[idx])) {
```

Comment:
@snowystinger - what do you think? Happy to go the `ref` way, if you prefer. Just to be fair, `some` here obviously loops through the elements, so that's another _O(n)_ in addition to the one inside `getBreakpointHandler`.

Reply:
I actually prefer this, I was picking up on you being worried about number of loops. If that wasn't what you were trying to communicate, sorry for misreading. I like this solution best, because as you pointed out, getBreakpointHandler is just a helper, and I'm really not worried about number of loops, this array will likely always be less than 10 items long (probably smaller)